controllers/argocd/status.go

```diff
@@ -250,16 +253,41 @@ func (r *ReconcileArgoCD) reconcileStatusNotifications(cr *argoprojv1a1.ArgoCD)
 func (r *ReconcileArgoCD) reconcileStatusHost(cr *argoprojv1a1.ArgoCD) error {
 	cr.Status.Host = ""
 	cr.Status.Phase = "Available"
-	if cr.Spec.Server.Route.Enabled {
-		if !IsRouteAPIAvailable() {
-			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
-			return nil
-		}
+
+	// Log an error if a user configures route on a platform where route API does not exist.
+	// This is a misconfiguration.
+	if cr.Spec.Server.Route.Enabled && !IsRouteAPIAvailable() {
+		err := errors.New("Misconfiguration:")
+		log.Error(err, "Routes not available in non-OpenShift environments, please use Ingresses instead")
+		return nil
+	}
```

Comment:
@iam-veeramalla Any specific reason the code for this check is placed here? Seems like https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/route.go#L217 might be a better place for it